src/x86.rs

```diff
@@ -86,7 +86,7 @@ impl Vector for __m64i {
     #[inline]
     #[target_feature(enable = "avx2")]
     unsafe fn loadu_si(a: *const Self) -> Self {
-        __m64i(_mm_loadu_si128(a as *const std::arch::x86_64::__m128i))
+        __m64i(_mm_loadl_epi64(a as *const __m128i))
```

Comment:
In this case, what are the contents of the other half of the `__m128i` returned? The [Intel docs](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadl_epi64&expand=3357) don't seem to say. I would be concerned about whether it's uninitialized or not.

Reply:
I looked here: https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadl_epi64&expand=3421,3357,3421,3343,3357
and it does seem to say that this is equivalent to:
```
dst[63:0] := MEM[mem_addr+63:mem_addr]
dst[MAX:64] := 0
```
What do you think?

Reply:
Additionally, I just looked at how the rust intrinsic is implemented at https://doc.rust-lang.org/src/core/up/up/stdarch/crates/core_arch/src/x86/sse2.rs.html#1232-1234:
```Rust
pub unsafe fn _mm_loadl_epi64(mem_addr: *const __m128i) -> __m128i {
    _mm_set_epi64x(0, ptr::read_unaligned(mem_addr as *const i64))
}
```
Maybe the code should just use this directly though?

Reply:
w.r.t. to Intel docs: derp. It is early here and I don't think I was fully awake. That code from the docs to me says that this is fine then! Sorry for the noise. :)

Reply:
Honestly this is all very fiddly. I wanted to use `_mm_loadu_si64` because technically the `loadu` should ensure unaligned load. However the code breaks when using this intrinsic. That's I why I then switched to `_mm_loadl_epi64`. There might be a big in how the Intrinsic is implemented in rust or in llvm. I try to do more testing later.

Reply:
Ah interesting. Yeah, might be worth filing a bug: https://github.com/rust-lang/stdarch/

Reply:
I have opened https://github.com/rust-lang/stdarch/issues/1166 for clarification. I might have totally misunderstood something though